examples/plot_2D_simulation_example.py

```diff
@@ -84,97 +150,194 @@ def plot(maps, titles, save_fig=False):
-    beta_hat, pval, pval_corr, one_minus_pval, one_minus_pval_corr = \
-        ensemble_clustered_inference(X_init, y, ward,
-                                     n_clusters, train_size=0.3)
-    zscore = zscore_from_pval(pval, one_minus_pval)
-    selected_ecdl = zscore > thr_c
-    selected_ecdl = np.logical_or(pval_corr < fwer_target / 2,
-                                  one_minus_pval_corr < fwer_target / 2)
-
-    maps = []
-    titles = []
-
-    maps.append(np.reshape(beta, shape))
-    titles.append('True weights')
-
-    maps.append(np.reshape(beta_extended, shape))
-    titles.append('True weights \nwith tolerance')
-
-    maps.append(np.reshape(selected_dl, shape))
-    titles.append('Desparsified Lasso')
-
-    maps.append(None)
-    titles.append(None)
-
-    maps.append(np.reshape(selected_cdl, shape))
-    titles.append('CluDL')
-
-    maps.append(np.reshape(selected_ecdl, shape))
-    titles.append('EnCluDL')
-
-    plot(maps, titles)
-
-
-if __name__ == '__main__':
-    main()
+##############################################################################
+# Generating the data
+# -------------------
+#
+# After setting the simulation parameters, we run the function that generates
+# the 2D scenario presented in introduction.
+
+# simulation parameters
+n_samples = 100
+shape = (40, 40)
+n_features = shape[1] * shape[0]
+roi_size = 4  # size of the edge of the four predictive regions
+sigma = 2.0  # noise standard deviation
+smooth_X = 1.0  # level of spatial smoothing introduced by the Gaussian filter
+
+# generating the data
+X_init, y, beta, epsilon, _, _ = \
+    multivariate_simulation(n_samples, shape, roi_size, sigma, smooth_X,
+                            seed=1)
+
+##############################################################################
+# Choosing inference parameters
+# -----------------------------
+#
+# The choice of the number of clusters depends on several parameters, such as:
+# the structure of the data (a higher correlation between neighboring features
+# enable a greater dimension reduction, i.e. a smaller number of clusters),
+# the number of samples (small datasets require more dimension reduction) and
+# the required spatial tolerance (small clusters lead to limited spatial
+# uncertainty). Formally, "spatial tolerance" is defined by the largest
+# distance from the true support for which the occurence of a false discovery
+# is not statistically controlled (c.f. References).
+# Theoretically, the spatial tolerance ``delta`` is equal to the largest
+# cluster diameter. However this choice is conservative, notably in the case
+# of ensembled clustered inference. For these algorithms, we recommend to take
+# the average cluster radius. In this example, we choose ``n_clusters = 200``,
+# leading to a theoretical spatial tolerance ``delta = 6``. However
+# following our pratical guidance ``delta = 2`` would have sufficient
+# for ensembled clustered inference algorithms (see Results).
```

Comment:
This last sentence is unclear. WDYM with "our practical guidance" ? Simply, "it turns out that..."